src/xcode/ENA/ENA/Source/HTTPClient/Service/Services/_Service+Default.swift

```diff
@@ -134,4 +149,115 @@ extension Service {
+		} else {
+			return serviceError
+		}
+	}
+	
+	/// Before failing, this checks if the resource has defined a defaultValue so we can call a .success with it instead of failing. If not defaultValue is given, it will fail as normal.
+	///
+	/// - Parameters:
+	///   - resource: Generic ("R") object and normally of type ReceiveResource.
+	///   - error: The error that would be thrown with the fail.
+	///   - completion: Swift-Result of loading. If successful, it contains the concrete object of our call.
+	func failureOrDefaultValueHandling<R>(
+		_ resource: R,
+		_ error: ServiceError<R.CustomError>,
+		_ completion: @escaping (Result<R.Receive.ReceiveModel, ServiceError<R.CustomError>>) -> Void
+	) where R: Resource {
+		if let defaultModel = resource.defaultModel {
+			Log.info("Found some default value", log: .client)
+			completion(.success(defaultModel))
+		} else {
+			Log.error("Found no default value. Will fail now.", log: .client, error: error)
+			completion(.failure(customError(in: resource, for: error)))
+		}
+	}
+
+	// MARK: - Private
+
+	/// checks if special cache handlings needs to be done. If looks up if cached data or default data can be returned.
+	/// Then the special use case handling is done otherwise the original error is given in completion handler
+	///
+	/// - Parameters:
+	///   - cacheUseCase: the caching use case that gets handled
+	///   - error: original error as a fallback for some cached use cases
+	///   - resource: Generic ("R") object and normally of type ReceiveResource.
+	///   - completion: Swift-Result of loading. If successful, it contains the concrete object of our call.
+	private func cacheUseCaseHandling<R>(
+		_ cacheUseCase: CacheUseCase,
+		_ error: Error?,
+		_ resource: R,
+		_ completion: @escaping (Result<R.Receive.ReceiveModel, ServiceError<R.CustomError>>) -> Void
+	) where R: Resource {
+		
+		// Check if we can handle caching use case handling
+		guard case let .caching(usage) = resource.type,
+			  usage.contains(cacheUseCase) else {
+				  // Otherwise, fall back to the default
+				  Log.info("No cache use case handling defined. Fallback to default error handling")
+				  defaultErrorHandling(error, resource, completion)
+				  return
+		}
+		
+		// If so, first we check if have something cached
+		if hasCachedData(resource) {
+			Log.info("Found some cached data", log: .client)
+			cached(resource, completion)
+			return
+		}
+		// If not, we will handle now the cache use cases
+		else {
+			Log.info("Found nothing cached. Handling cache use cases")
+			switch cacheUseCase {
+			case .noNetwork:
+				defaultErrorHandling(error, resource, completion)
+			case .statusCode(let statusCodes):
+				Log.error("Unexpected server error: (\(statusCodes)", log: .client)
+				completion(.failure(customError(in: resource, for: .unexpectedServerError(statusCodes))))
+			}
+		}
+	}
+	
+	/// Handle default error for no network. If the error is nil, it is an invalid response.
```

Comment:
Honestly I don't Unterstand the error handling only by looking at this function. It would be hard to maintain. 
Why is it an invalid response if the error is nil?
And why is it a transportationError in other cases?